<launch>
    <arg name="drone_id" value="0"/>
    <arg name="rviz" default="true"/>
    <arg name="drone_num" default="2"/>

    <node name="drone_$(arg drone_id)_v2l_registration_node" pkg="registration" type="visual2lidar_registration_node" output="screen" >
        <!-- rostpic remap -->
        <remap from="~full_map" to="/Map_old"/>

        <remap from="~Odometry_src" to="/corrected_imu_propagate_odom"/>
        <remap from="~Map_src" to="/moving_map_node/grid_map/occupancy"/>

        <remap from="~Odometry_modify" to="/Odom_new"/>
        <remap from="~Map_modify" to="/Map_new"/>

        <param name="drone_id" value="$(arg drone_id)"/>

        <!-- full map point clouds params -->
        <rosparam param="map_size">[-10, 30, -60, 60, -2, 20]</rosparam>
        <rosparam param="cut_globalmap_size">[-8, 8, -8, 8, -1.0, 5]</rosparam>
        <param name="is_use_pcd"            value="true"/>
        <param name="fullmap_fn"           value="/home/zuzu/Documents/BAG/FY_Project/rs_lidar/uwb/out.pcd"/>  
        <param name="resolution"            value="0.1"/>
        <param name="radiusfilter_radius"    value="0.15"/>
        <param name="radiusfilter_neibors"  value="5"/>

        <!-- registration params -->
        <param name="reg_pattern"       value="pclGICP"/>
        <rosparam param="init_guess">[0,0,0,0,0,0]</rosparam>
        <param name="reg_iter"          value="5"/>
        <param name="trans_eps"         value="1e-2"/>
        <param name="fitness_eps"       value="1e-2"/>
        <param name="max_cor_dis"       value="5"/>        
        <param name="ransac_iter"       value="20"/>        
        <param name="neibor_k"          value="5"/>
        <param name="ndt_resolution"    value="0.5"/>
        <param name="ndt_stepsize"      value="0.8"/>

        <!-- evaluate params -->
        <param name="ground_height" value="0.15"/>
        <param name="socre_thre"    value="0.15"/>


    </node>

	<group if="$(arg rviz)">
        <node name="rviz" pkg="rviz" type="rviz" args="-d $(find registration)/config/v2l.rviz" required="true"/>
	</group>
  
  </launch>
  
